File: 7_kyu/Speed_Control.go
package kata

func Gps(s int, x []float64) int {
	var max_delta float64
	for index := 1; index < len(x); index++ {
		delta := x[index] - x[index-1]
		if max_delta < delta {
			max_delta = delta
		}
	}
	return int(3600*max_delta) / s
}

File: 7_kyu/Speed_Control.rs
fn gps(s: i32, x: Vec<f64>) -> i32 {
    let max_delta = x.windows(2).map(|x| x[1] - x[0]).fold(0., f64::max);
    let max_average_speed = 3600. * max_delta / s as f64;
    max_average_speed as i32
}
